pymfe/mfe.py

```diff
@@ -1727,21 +1832,50 @@ def extract_with_confidence(
         if return_avg_val:
             mtf_vals = np.nanmean(mtf_vals, axis=1)
 
-        if self.timeopt:
-            if return_avg_val:
-                mtf_time /= sample_num
+        if self.timeopt and return_avg_val:
+            mtf_time /= sample_num
+
+        _deal_types = {
+            tuple: lambda names, vals, conf, times = []:
+                (names, vals, times, conf) if self.timeopt
+                else (names, vals, conf),
+            dict: lambda names, vals, conf, times = []:
+                {
+                    "mtf_names": names,
+                    "mtf_vals": vals,
+                    "confidance": conf,
+                    "mtf_time": times
+                } if self.timeopt else
+                {
+                    "mtf_names": names,
+                    "mtf_vals": vals,
+                    "confidance": conf
```

Comment:
Here "confidance" looks misspelled (not being picky on typos, but since in this case it is a dictionary key it will probably affect users directly).

Reply:
True! My bad.
I think it is fixed now :)

Reply:
Thanks for the review!